src/components/UI/UpdateComponent/index.tsx

```diff
@@ -1,18 +1,84 @@
-import './index.css'
-
-import React from 'react'
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faSyncAlt } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import cx from 'classnames'
+import React, { useEffect, useMemo, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import './index.css'
 
-type Props = {
+export interface UpdateComponentProps {
   message?: string
+  inline?: boolean
 }
 
-export default function UpdateComponent({ message = 'Loading' }: Props) {
+export function UpdateComponentBase({
+  message = 'Loading',
+  inline = false
+}: UpdateComponentProps) {
+  const componentRef = useRef<HTMLDivElement>(null)
+  const wrapperRef = useRef<HTMLDivElement>(null)
+  const observer = useMemo(() => {
+    return new IntersectionObserver(
+      (entries) => {
+        if (wrapperRef.current) {
+          for (const entry of entries) {
+            const ratio =
+              entry.intersectionRect.height / entry.boundingClientRect.height
+            wrapperRef.current.style.height = `${100 * ratio}%`
+          }
+        }
+      },
+      {
+        threshold: Array(100)
+          .fill(0)
+          .map((_, i) => (i + 1) / 100)
+      }
+    )
+  }, [wrapperRef.current])
+
+  useEffect(() => {
+    return () => observer.disconnect()
+  }, [observer])
+
+  useEffect(() => {
+    if (inline) {
+      const component = componentRef.current
+      if (component) {
+        observer.observe(component)
+        return () => {
+          observer.unobserve(component)
+        }
+      }
+    }
+    return
+  }, [inline, observer, componentRef.current])
+
   return (
-    <div className="updateIcon" data-testid="updateComponent">
-      <FontAwesomeIcon className="icon" icon={faSyncAlt} />
-      <span>{message}</span>
+    <div
+      className={cx('UpdateComponent', {
+        'UpdateComponent--inline': inline,
+        'UpdateComponent--topLevel': !inline
+      })}
+      data-testid="updateComponent"
+      ref={componentRef}
+    >
+      <div
+        className="UpdateComponent__wrapper"
+        data-testid="updateComponent"
+        ref={wrapperRef}
+      >
+        <FontAwesomeIcon className="UpdateComponent__icon" icon={faSyncAlt} />
+        {message && <div>{message}</div>}
+      </div>
     </div>
   )
 }
+export default function UpdateComponent({
+  message,
+  ...rest
+}: UpdateComponentProps) {
+  const { t } = useTranslation()
+  if (message === undefined) {
```

Comment:
I believe a better check would be:
`if (!message) { `

because then it would use the default also if we send an empty string, null or any other falsy value.

Reply:
That's deliberate:
a) It's inline with current behavior because default value only work if the property is missing (`undefined`). You can pass empty string to the component and it won't display default message.
b) I can't see any reason to restrict ability to pass empty string and display icon without any message
c) You can't pass any falsy value, because type checking won't allow it. You can either pass no value, which will cause displaying default value or empty string which will display no message

Reply:
Ah, I see. Makes sense to load only the icon.
Good then.